lib/Form/DataTransformer/DateIntervalToArrayTransformer.php

```diff
@@ -0,0 +1,180 @@
+<?php
+/**
+ * This file is part of the eZ RepositoryForms package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+
+namespace EzSystems\RepositoryForms\Form\DataTransformer;
+
+use DateInterval;
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+/**
+ * Data transformer from PHP DateInterval to array for form inputs
+ */
+class DateIntervalToArrayTransformer implements DataTransformerInterface
+{
+    /**
+     * @var bool If false, will remove leading zeros from datetime elements
+     */
+    private $pad;
+
+    private $fields;
+
+    /**
+     * Constructor.
+     *
+     * @param array  $fields         The date fields
+     * @param bool   $pad            Whether to use padding
+     */
+    public function __construct(array $fields = null, $pad = false)
+    {
+        if (null === $fields) {
+            $fields = [ 'year', 'month', 'day', 'hour', 'minute', 'second' ];
+        }
+
+        $this->fields = $fields;
+        $this->pad = (bool)$pad;
+    }
+
+    /**
+     * Transforms a date interval into an array of date interval elements
+     *
+     * @param DateInterval $dateInterval Date interval.
+     *
+     * @return array Date interval elements.
+     *
+     * @throws TransformationFailedException If the given value is not an instance of DateInterval
+     */
+    public function transform($dateInterval)
```

Comment:
I find this one quite complicated... It looks really cool, but my brains are bleeding a little bit.

I think it could either be commented a bit (that's level 1), or splitted into a couple functions with well named functions.